docs/OneBlinkForm.md

```diff
@@ -0,0 +1,274 @@
+
+| Property     | Type           | Description                                                                                                                       |
+| ------------ | -------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| `submission` | `Object`       | The submission data                                                                                                               |
+| `definition` | `OneBlinkForm` | The OneBlink Form, this will be different from the `form` prop passed to the Component as it is cloned when the component mounts. |
+
+### NewFormSubmission
+
+Inherits properties from [`NewDraftSubmission`](#newdraftsubmission)
+
+| Property        | Type       | Description                                         |
+| --------------- | ---------- | --------------------------------------------------- |
+| `captchaTokens` | `string[]` | Captcha tokens gathered by a `captcha` Form Element |
+
+### Example
+
+```js
+// @flow
+
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { BrowserRouter, Route, useHistory } from 'react-router-dom'
+import {
+  OneBlinkAppsError,
+  draftService,
+  submissionService,
+} from '@oneblink/apps'
+import '@oneblink/apps-react/dist/styles.css'
+import {
+  IsOfflineContextProvider,
+  OneBlinkForm,
+  useIsMounted,
+} from '@oneblink/apps-react'
+
+const formsAppId = 1
+const form: Form = {
+  id: 1,
+  name: 'Name of Form',
+  description: '',
+  organisationId: 'abc123',
+  formsAppEnvironmentId: 1,
+  formsAppIds: [],
+  elements: [],
+  isAuthenticated: false,
+  isMultiPage: false,
+  isInfoPage: false,
+  publishStartDate: null,
+  publishEndDate: null,
+  postSubmissionAction: 'FORMS_LIBRARY',
+  submissionEvents: [],
+  tags: [],
+}
+
+function FormContainer() {
+  const isMounted = useIsMounted()
+  const history = useHistory()
+  const [
+    postSubmissionActionErrorMessage,
+    setPostSubmissionActionErrorMessage,
+  ] = React.useState<string>('')
+  const [{ isSavingDraft, saveDraftError }, setSaveDraftState] = React.useState(
+    {
+      isSavingDraft: false,
+      saveDraftError: null,
+    },
+  )
+
+  const [
+    { isSubmitting, submitError, formSubmissionResult },
+    setSubmitState,
+  ] = React.useState({
+    formSubmissionResult: null,
+    isSubmitting: false,
+    submitError: null,
+  })
+
+  const handlePostSubmissionAction = React.useCallback(
+    async (formSubmissionResult: FormSubmissionResult) => {
+      try {
+        await submissionService.executePostSubmissionAction(
+          formSubmissionResult,
+          history.push,
+        )
+      } catch (error) {
+        if (isMounted.current) {
+          setPostSubmissionActionErrorMessage(error.message)
+        }
+      }
+    },
+    [history.push, isMounted],
+  )
+
+  const handleSubmit = React.useCallback(
+    async (formSubmission: FormSubmission) => {
```

Comment:
These typings won't be valid to anyone not using our flow types (no one?). We could `import { FormTypes } from '@oneblink/apps' and then use `FormTypes.FormSubmission` to make the code valid ts?

Reply:
Sounds good.